src/notifications/kernel.rs

```diff
@@ -0,0 +1,125 @@
+use std::ffi::{c_char, c_void};
```

Comment:
In commit a0c949f7526221e45b03c48286eabc33f70099c7: Why not just call this file `notifications.rs` too?

Reply:
Looks good.